code/modules/mob/mob_emote.dm

```diff
@@ -0,0 +1,192 @@
+	for(var/key in GLOB.emote_list)
+		for(var/datum/emote/P in GLOB.emote_list[key])
+			if((P.key in all_keys))
+				continue
+			if(P.can_run_emote(src, status_check = FALSE, intentional = null))
+				all_keys += P.key
+				if(P.key_third_person)
+					all_keys += P.key_third_person
+
+	return all_keys
+
+/datum/emote/help
+	key = "help"
+	mob_type_ignore_stat_typecache = list(/mob/dead/observer, /mob/living/silicon/ai)
+
+/datum/emote/help/run_emote(mob/user, params, type_override, intentional)
+	. = ..()
+	var/list/base_keys = list()
+	var/list/all_keys = list()
+	var/list/species_emotes = list()
+	var/list/message = list("Available emotes, you can use them with say \"*emote\": ")
+
+	var/mob/living/carbon/human/H = user
+	for(var/key in GLOB.emote_list)
+		for(var/datum/emote/P in GLOB.emote_list[key])
+			var/full_key = P.key
+			if((P.key in all_keys))
+				continue
+			if(P.can_run_emote(user, status_check = FALSE, intentional = TRUE))
+				if(P.message_param && P.param_desc)
+					// Add our parameter description, like flap-user
+					full_key = P.key + "\[[EMOTE_PARAM_SEPARATOR][P.param_desc]\]"
+				if(istype(H) && P.species_type_whitelist_typecache && H.dna && is_type_in_typecache(H.dna.species, P.species_type_whitelist_typecache))
+					species_emotes += full_key
+				else
+					base_keys += full_key
+				all_keys += P.key
+
+	base_keys = sortList(base_keys)
+
+	message += base_keys.Join(", ")
+	message += "."
+	message = message.Join("")
+	if(length(species_emotes) > 0)
+		species_emotes = sortList(species_emotes)
+		message += "\n<u>[user?.dna?.species.name] specific emotes</u> :- "
+		message += species_emotes.Join(", ")
+		message += "."
+	to_chat(user, message)
+
+/datum/emote/flip
+	key = "flip"
+	key_third_person = "flips"
+	message = "does a flip!"
+	hands_use_check = TRUE
+	emote_type = EMOTE_VISIBLE | EMOTE_FORCE_NO_RUNECHAT  // don't need an emote to see that
+	mob_type_allowed_typecache = list(/mob/living, /mob/dead/observer)  // okay but what if we allowed ghosts to flip as well
+	mob_type_blacklist_typecache = list(/mob/living/carbon/brain, /mob/camera, /mob/living/silicon/ai)
+	mob_type_ignore_stat_typecache = list(/mob/dead/observer)
+
+/datum/emote/flip/run_emote(mob/user, params, type_override, intentional)
+
+	if(isobserver(user))
+		user.SpinAnimation(5, 1)
+		return TRUE
+
+	var/mob/living/L = user
+
+	if(user.lying || (istype(L) && L.IsWeakened()))
+		message = "flops and flails around on the floor."
+	else if(params)
+		message_param = "flips in %t's general direction."
+	else if(ishuman(user))
+		var/mob/living/carbon/human/H = user
+		if(istype(H.get_active_hand(), /obj/item/grab))
+			var/obj/item/grab/G = H.get_active_hand()
+			if(G && G.affecting)
+				if(H.buckled || G.affecting.buckled)
+					to_chat(user, "<span class='warning'>[G.affecting] is buckled, you can't flip around [G.affecting.p_them()]!</span>")
+					return TRUE
+				var/turf/oldloc = user.loc
+				var/turf/newloc = G.affecting.loc
```

Comment:
`get_turf()` maybe?

Reply:
I think this has merit as it is, since we check after to see if they're on a turf or not (probably don't want them moving if they've somehow been grabbed in a closet or something).